projects/angular-cld/src/lib/cloudinary-sdk-analytics.spec.ts

```diff
@@ -0,0 +1,215 @@
+
+      // all elements with an attached CloudinaryImage
+      des = fixture.debugElement.query(By.directive(CloudinaryImage));
+    });
+
+    it('creates an img without encoding', () => {
+      const img = des.children[0].nativeElement as HTMLImageElement;
+      expect(img.attributes.getNamedItem('src').value).toEqual('http://res.cloudinary.com/@@fake_angular_sdk@@/image/upload/sample');
+    });
+  });
+
+  describe('Config with analytics set to true', () => {
+    let localCloudinary: Cloudinary = new Cloudinary(require('cloudinary-core'),
+      { cloud_name: '@@fake_angular_sdk@@', analytics: true} as CloudinaryConfiguration);
+
+    beforeEach(() => {
+      spyOn(localCloudinary, 'toCloudinaryAttributes').and.callThrough();
+      spyOn(localCloudinary, 'url').and.callThrough();
+    });
+    @Component({
+      template: `<cl-image public-id="sample"></cl-image>`
+    })
+    class TestComponent { }
+
+    let fixture: ComponentFixture<TestComponent>;
+    let des: DebugElement;  // the elements w/ the directive
+
+    beforeEach(() => {
+      fixture = TestBed.configureTestingModule({
+        declarations: [CloudinaryImage, TestComponent],
+        providers: [{ provide: Cloudinary, useValue: localCloudinary }]
+      }).createComponent(TestComponent);
+
+      fixture.detectChanges(); // initial binding
+
+      // all elements with an attached CloudinaryImage
+      des = fixture.debugElement.query(By.directive(CloudinaryImage));
+    });
+
+    it('creates an img with feature defaulting to 0', () => {
+      const img = des.children[0].nativeElement as HTMLImageElement;
+      expect(img.attributes.getNamedItem('src').value).toEqual('http://res.cloudinary.com/@@fake_angular_sdk@@/image/upload/sample?_s=KE7pBq0');
+    });
+  });
+
+  describe('Accessibility with analytics set to true', () => {
+    let localCloudinary: Cloudinary = new Cloudinary(require('cloudinary-core'),
+      { cloud_name: '@@fake_angular_sdk@@', analytics: true} as CloudinaryConfiguration);
+
+    beforeEach(() => {
+      spyOn(localCloudinary, 'toCloudinaryAttributes').and.callThrough();
+      spyOn(localCloudinary, 'url').and.callThrough();
+    });
+    @Component({
+      template: `<cl-image accessibility="darkmode" public-id="sample"></cl-image>`
+    })
+    class TestComponent { }
+
+    let fixture: ComponentFixture<TestComponent>;
+    let des: DebugElement;  // the elements w/ the directive
+
+    beforeEach(() => {
+      fixture = TestBed.configureTestingModule({
+        declarations: [CloudinaryImage, TestComponent],
+        providers: [{ provide: Cloudinary, useValue: localCloudinary }]
+      }).createComponent(TestComponent);
+
+      fixture.detectChanges(); // initial binding
+
+      // all elements with an attached CloudinaryImage
+      des = fixture.debugElement.query(By.directive(CloudinaryImage));
+    });
+
+    it('creates an img with feature defaulting to D', () => {
```

Comment:
Why does the feature "default" to D? 
(again, implying nothing is passed, and the default value is D)

maybe there's a better name for it?